tests/foreman/cli/test_discoveryrule.py

```diff
@@ -34,68 +35,70 @@
 from robottelo.cli.user import User
 from robottelo.datafactory import filtered_datapoint
 from robottelo.datafactory import invalid_values_list
+from robottelo.datafactory import parametrized
 from robottelo.datafactory import valid_data_list
-from robottelo.test import CLITestCase
 
 
-class DiscoveryRuleTestCase(CLITestCase):
-    """Implements Foreman discovery Rules tests in CLI."""
+@pytest.fixture(scope='class')
+def class_hostgroup(class_org, class_location):
+    return make_hostgroup(
```

Comment:
Agreed, I'd like to leave that to the component owner or a concerted effort to optimize and consolidate the fixtures for basic entities.

Reply:
During debugging, I may have changed my mind.

The CLI factory is returning dictionaries, not actual objects. :-1: 

This means fixtures provided by API have attributized access against their objects, and this hostgroup would be a dictionary that would require either resolving an instance by ID lookup, or using key access.

I'll likely change this to be an API based fixture, so that only the discoveryrule created by the `discoveryrule_factory` is done through CLI (as its the target of the test coverage)